libsolidity/analysis/ImmutableValidator.cpp

```diff
@@ -122,6 +122,12 @@ bool ImmutableValidator::visit(WhileStatement const& _whileStatement)
 	return false;
 }
 
+void ImmutableValidator::endVisit(IdentifierPath const& _identifierPath)
+{
+	if (auto const callableDef = dynamic_cast<CallableDeclaration const*>(_identifierPath.annotation().referencedDeclaration))
+		visitCallableIfNew(callableDef->resolveVirtual(m_currentContract));
```

Comment:
I'm actually wondering if this really needs virtual lookup in all cases. If you write `C.f` where `C` is a contract name, then virtual lookup should not be performed, but if you write `f` then it should.

Reply:
Also, this can refer to a variable, I think (through a contract name)

Reply:
Can you please check this?

Reply:
Wouldn't `C.f` not be a `MemberAccess` though?

Reply:
Checking the parser again, an `IdentifierPath` can only be in an `InheritanceSpecifier`, `OverrideSpecifier`, `UserDefinedTypeName`, `ModifierInvocation` and `UsingDirective`, so I don't actually think it can be a variable nor can it be the `C.f` situation at least for a normal call in statements only maybe as a modifier.